### PR TITLE
fix: get pixels method

### PR DESCRIPTION
**Pull request by seixasxbr**
# What is this PR for
[sc-46514]
fix: fix the get pixels method
The second element of the pair was pointing to the last element in the radar echo dot, not to the end of the range

chore: use the get pixels method inside the update image

# Checklist
- [x] Assign yourself  in the PR
- [x] Write unit tests (when relevant)

### CODE REVIEW

File: test/test_EchoToImageLUT.cpp [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include <gtest/gtest.h>
 #include <radar_base/EchoToImageLUT.hpp>
 
+#include <iostream>
+
```
Comment (wvmcastro) [inline test/test_EchoToImageLUT.cpp]:
```suggestion
```

File: src/EchoToImageLUT.hpp [hunk 2]
```diff
@@ -34,9 +34,10 @@ namespace radar_base {
     public:
         EchoToImageLUT() = delete;
         /**
-         * @brief Returns a pair of iterators pointing to the initial and the final pixel
-         * in a radar echo dot represented by an angle index and a sweep index at the
-         * echoes to image look-up table
+         * @brief Returns a pair of constant iterators defining a range of pixels in a
+         * radar echo dot, determined by an angle index and a sweep index, at the echoes
+         * to image look-up table
```
Comment (wvmcastro) [inline src/EchoToImageLUT.hpp]:
```suggestion
         * radar echo dot, determined by an angle index and a echo dot index, at the echoes
         * to image look-up table
```

File: src/EchoToImageLUT.cpp [hunk 3]
```diff
@@ -134,8 +130,7 @@ void EchoToImageLUT::drawImageFromEchoes(std::vector<uint8_t> const& world_echoe
 pair<vector<Point>::const_iterator, vector<Point>::const_iterator> EchoToImageLUT::
     getPixels(int angle_idx, int sweep_idx) const
 {
-    int initial_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx];
-    int final_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1] - 1;
-    return std::make_pair(m_data.cbegin() + initial_point_idx,
-        m_data.cbegin() + final_point_idx);
+    int pixels_begin = m_data_index[angle_idx * m_sweep_size + sweep_idx];
+    int pixels_end = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1];
```
Comment (wvmcastro) [inline src/EchoToImageLUT.cpp]: It is not actually sweep index, is the echo dot index
```suggestion
    int pixels_begin = m_data_index[angle_idx * m_sweep_size + echo_dot_idx];
    int pixels_end = m_data_index[angle_idx * m_sweep_size + echo_dot_idx + 1];
```

File: src/EchoToImageLUT.hpp (hunk 2 above)
Comment (seixasxbr) [reply src/EchoToImageLUT.hpp:39]: eho_dot_idx = angle_idx * sweep_length + sweep_idx

File: src/EchoToImageLUT.cpp (hunk 3 above)
Comment (seixasxbr) [reply src/EchoToImageLUT.cpp:134]: IMO it is indeed is the sweep idx. The range is [0, sweep_length[

**Review by wvmcastro — APPROVED** [review]

**Review by kapeps — APPROVED** [review]

**Pull request merged**